scheduler/feasible.go

```diff
@@ -381,6 +381,11 @@ func (c *NetworkChecker) SetNetwork(network *structs.NetworkResource) {
 }
 
 func (c *NetworkChecker) Feasible(option *structs.Node) bool {
+	// Allow jobs not requiring any network resources
+	if c.ports == nil {
+		return true
+	}
```

Comment:
There's a [`network.mode = "none"`](https://www.nomadproject.io/docs/job-specification/network#none) that I would expect to be set in the scenario you've described. Rather than using nil ports as an implied "no network", it'd be better if we checked explicitly on that mode. That way we're not confusing potential misconfiguration for an intentional user instruction.